cloudops.go

```diff
@@ -82,10 +82,12 @@ type InstanceInfo struct {
 type Compute interface {
 	// InstanceID of instance where command is executed.
 	InstanceID() string
-	// Inspect instance identified by ID.
-	InspectIntance(ID string) (*InstanceInfo, error)
-	// InspectInstanceGroup returns instanceGroupInfo matching labels.
-	InspectInstanceGroup(labels map[string]string) (*InstanceInfo, error)
+	// InspectSelf inspects the node where the code is invoked. Hence this assumes
+	// this is invoked from within a cloud instance
+	InspectSelf() (*InstanceInfo, error)
+	// InspectSelfInstanceGroup inspects the instance group within which the current
+	// instance resides
```

Comment:
There is no need for the assumption that it is invoked from the same instance.   Take the InstanceID as a parameter to allow cloudops to be run as n API server from _an_ instance that has the _right_ privileges.

Reply:
Done. Updated the compute APIs to

```
        // InspectSelf inspects the node where the code is invoked.
        InspectInstance(instanceID string) (*InstanceInfo, error)
        // InspectSelfInstanceGroupForInstance inspects the instance group to which
        // the cloud instance with given ID belongs
        InspectInstanceGroupForInstance(instanceID string) (*InstanceGroupInfo, error)
```

Used InspectInstanceGroupForInstance since our use case is to inspect the group by instance instead of a generic label query.